File: app/src/main/java/com/bitrise/app/ui/adapters/list/models/CategoryItem.java
package com.bitrise.app.ui.adapters.list.models;

public class CategoryItem implements GenericCategoryItem<String> {

    public String category;

    public CategoryItem(String category) {
        this.category = category;
    }

    public String getData() {
        return this.category;
    }

    public int getType() {
        return 1004;
    }

    public String getCategoryName() {
        return this.category;
    }

    public int compareTo(GenericCategoryItem categoryItem) {
        return this.category.compareTo(categoryItem.getCategoryName());
    }
}

File: app/src/main/java/com/bitrise/app/ui/adapters/list/models/CategoryItem.kt
package com.bitrise.app.ui.adapters.list.models

import com.bitrise.app.ui.factories.ITEM_LIST_TYPE_CATEGORY

class CategoryItem(
    override var data: CharSequence,
    override var weight: Int,
) : GenericCategoryItem<CharSequence> {

    override val type: Int
        get() = ITEM_LIST_TYPE_CATEGORY

    override val categoryName: String
        get() = data.toString()

    override fun compareTo(categoryItem: GenericCategoryItem<*>): Int {
        return categoryName.compareTo(categoryItem.categoryName)
    }
}